apps/prairielearn/src/middlewares/selectAndAuthzInstanceQuestion.ts

```diff
@@ -1,24 +1,82 @@
 import { type Request, type Response } from 'express';
 import asyncHandler from 'express-async-handler';
+import z from 'zod';
 
 import * as error from '@prairielearn/error';
 import * as sqldb from '@prairielearn/postgres';
 
+import {
+  AssessmentInstanceSchema,
+  AssessmentQuestionSchema,
+  AssessmentSchema,
+  AssessmentSetSchema,
+  AuthzAssessmentInstanceSchema,
+  FileSchema,
+  GroupSchema,
+  InstanceQuestionSchema,
+  NextAllowedGradeSchema,
+  QuestionSchema,
+  UserSchema,
+} from '../lib/db-types.js';
 import { getGroupConfig, getGroupInfo, getQuestionGroupPermissions } from '../lib/groups.js';
 
 const sql = sqldb.loadSqlEquiv(import.meta.url);
 
+const InstanceQuestionInfoSchema = z.object({
+  id: z.number(),
```

Comment:
Shouldn't this be `IdSchema`? Ditto for other IDs here.

Reply:
Done in cc7a398202e4f95f628d1abb36a7b720cd21bfbe.